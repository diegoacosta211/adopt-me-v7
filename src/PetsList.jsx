import Pet from './Pet';
const PetsList = ({ pets }) => (
  <div>
    {!pets.length ? (
      <h2>No Pets Found!</h2>
    ) : (
      pets.map(({ id, name, breed, animal, images, city, state }) => (
        <Pet
          key={id}
          id={id}
          name={name}
          breed={breed}
          animal={animal}
          images={images}
          location={`${city}, ${state}`}
        />
      ))
    )}
  </div>
);

export default PetsList;
