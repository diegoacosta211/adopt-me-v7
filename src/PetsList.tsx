import { FunctionComponent } from 'react';
import Pet from './Pet';
import { Pet as PetType} from './types';

interface IProps {
  pets: PetType[]
}

const PetsList: FunctionComponent<IProps> = ({ pets }) => (
  <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-4">
    {!pets.length ? (
      <h2>No Pets Found!</h2>
    ) : (
      pets.map(({ id, name, breed, animal, images, city, state }) => (
        <Pet
          key={id}
          id={id}
          name={name}
          breed={breed}
          animal={animal}
          images={images}
          location={`${city}, ${state}`}
        />
      ))
    )}
  </div>
);

export default PetsList;
