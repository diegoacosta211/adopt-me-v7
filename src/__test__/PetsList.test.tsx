import { expect, test } from '@jest/globals';
import { render } from '@testing-library/react';
import { create } from 'react-test-renderer';
import { createRenderer } from 'react-test-renderer/shallow';
import PetsList from '../PetsList';
import { Animal, Pet } from '../types';

// sample
const pets = [
  {
    id: 1,
    name: "Luna",
    animal: "dog",
    city: "Seattle",
    state: "WA",
    description:
      "Luna is actually the most adorable dog in the world. Her hobbies include yelling at squirrels, aggressively napping on her owners' laps, and asking to be fed two hours before IT'S DAMN WELL TIME LUNA. Luna is beloved by her puppy parents and lazily resides currently in Seattle, Washington.",
    breed: "Havanese",
    images: [
      "http://pets-images.dev-apis.com/pets/dog25.jpg",
      "http://pets-images.dev-apis.com/pets/dog26.jpg",
      "http://pets-images.dev-apis.com/pets/dog27.jpg",
      "http://pets-images.dev-apis.com/pets/dog28.jpg",
      "http://pets-images.dev-apis.com/pets/dog29.jpg",
    ],
  },
  {
    id: 2,
    name: "Charisse",
    animal: "rabbit",
    city: "Lexington",
    state: "KY",
    description:
      "Maecenas leo odio, condimentum id, luctus nec, molestie sed, justo. Pellentesque viverra pede ac diam. Cras pellentesque volutpat dui.\n\nMaecenas tristique, est et tempus semper, est quam pharetra magna, ac consequat metus sapien ut nunc. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Mauris viverra diam vitae quam. Suspendisse potenti.",
    breed: "Havanese",
    images: ["http://pets-images.dev-apis.com/pets/rabbit0.jpg"],
  },
  {
    id: 3,
    name: "Maitilde",
    animal: "rabbit",
    city: "Dallas",
    state: "TX",
    description:
      "Duis consequat dui nec nisi volutpat eleifend. Donec ut dolor. Morbi vel lectus in quam fringilla rhoncus.\n\nMauris enim leo, rhoncus sed, vestibulum sit amet, cursus id, turpis. Integer aliquet, massa id lobortis convallis, tortor risus dapibus augue, vel accumsan tellus nisi eu orci. Mauris lacinia sapien quis libero.\n\nNullam sit amet turpis elementum ligula vehicula consequat. Morbi a ipsum. Integer a nibh.",
    breed: "Lab",
    images: ["http://pets-images.dev-apis.com/pets/rabbit1.jpg"],
  },
  {
    id: 4,
    name: "Natalina",
    animal: "rabbit",
    city: "Tampa",
    state: "FL",
    description:
      "Mauris enim leo, rhoncus sed, vestibulum sit amet, cursus id, turpis. Integer aliquet, massa id lobortis convallis, tortor risus dapibus augue, vel accumsan tellus nisi eu orci. Mauris lacinia sapien quis libero.\n\nNullam sit amet turpis elementum ligula vehicula consequat. Morbi a ipsum. Integer a nibh.\n\nIn quis justo. Maecenas rhoncus aliquam lacus. Morbi quis tortor id nulla ultrices aliquet.",
    breed: "Lab",
    images: ["http://pets-images.dev-apis.com/pets/rabbit2.jpg"],
  },
  {
    id: 5,
    name: "Michail",
    animal: "reptile",
    city: "Tuscaloosa",
    state: "AL",
    description:
      "Nulla ut erat id mauris vulputate elementum. Nullam varius. Nulla facilisi.\n\nCras non velit nec nisi vulputate nonummy. Maecenas tincidunt lacus at velit. Vivamus vel nulla eget eros elementum pellentesque.\n\nQuisque porta volutpat erat. Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla. Nunc purus.",
    breed: "Havanese",
    images: ["http://pets-images.dev-apis.com/pets/reptile1.jpg"],
  },
] as Pet[];


test("renders correctly with no pets", () => {
  const tree = create(<PetsList pets={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders correctly with some pets", () => {
  const r = createRenderer();
  r.render(<PetsList pets={pets} />);
  expect(r.getRenderOutput()).toMatchSnapshot();
})

