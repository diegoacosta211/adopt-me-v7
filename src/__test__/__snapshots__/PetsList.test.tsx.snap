// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no pets 1`] = `
<div
  className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-4"
>
  <h2>
    No Pets Found!
  </h2>
</div>
`;

exports[`renders correctly with some pets 1`] = `
<div
  className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-4"
>
  <Pet
    animal="dog"
    breed="Havanese"
    id={1}
    images={
      Array [
        "http://pets-images.dev-apis.com/pets/dog25.jpg",
        "http://pets-images.dev-apis.com/pets/dog26.jpg",
        "http://pets-images.dev-apis.com/pets/dog27.jpg",
        "http://pets-images.dev-apis.com/pets/dog28.jpg",
        "http://pets-images.dev-apis.com/pets/dog29.jpg",
      ]
    }
    location="Seattle, WA"
    name="Luna"
  />
  <Pet
    animal="rabbit"
    breed="Havanese"
    id={2}
    images={
      Array [
        "http://pets-images.dev-apis.com/pets/rabbit0.jpg",
      ]
    }
    location="Lexington, KY"
    name="Charisse"
  />
  <Pet
    animal="rabbit"
    breed="Lab"
    id={3}
    images={
      Array [
        "http://pets-images.dev-apis.com/pets/rabbit1.jpg",
      ]
    }
    location="Dallas, TX"
    name="Maitilde"
  />
  <Pet
    animal="rabbit"
    breed="Lab"
    id={4}
    images={
      Array [
        "http://pets-images.dev-apis.com/pets/rabbit2.jpg",
      ]
    }
    location="Tampa, FL"
    name="Natalina"
  />
  <Pet
    animal="reptile"
    breed="Havanese"
    id={5}
    images={
      Array [
        "http://pets-images.dev-apis.com/pets/reptile1.jpg",
      ]
    }
    location="Tuscaloosa, AL"
    name="Michail"
  />
</div>
`;
